packages/js-dpp/lib/identity/stateTransition/IdentityUpdateTransition/validation/state/validateIdentityUpdateTransitionStateFactory.js

```diff
@@ -0,0 +1,108 @@
+const ValidationResult = require('../../../../../validation/ValidationResult');
+const IdentityNotFoundError = require('../../../../../errors/consensus/state/identity/IdentityNotFoundError');
+const InvalidIdentityRevisionError = require('../../../../../errors/consensus/state/identity/InvalidIdentityRevisionError');
+const IdentityPublicKey = require('../../../../IdentityPublicKey');
+const IdentityPublicKeyIsReadOnlyError = require('../../../../../errors/consensus/state/identity/IdentityPublicKeyIsReadOnlyError');
+const MissedSecurityLevelIdentityPublicKeyError = require('../../../../../errors/consensus/state/identity/MissedSecurityLevelIdentityPublicKeyError');
+
+// security levels which must have at least one key
+const SECURITY_LEVELS = [
+  IdentityPublicKey.SECURITY_LEVELS.MASTER,
+];
+
+/**
+ * @param {StateRepository} stateRepository
+ * @param {validatePublicKeys} validatePublicKeys
+ * @return {validateIdentityUpdateTransitionState}
+ */
+function validateIdentityUpdateTransitionStateFactory(
+  stateRepository,
+  validatePublicKeys,
+) {
+  /**
+   * @typedef {validateIdentityUpdateTransitionState}
+   * @param {IdentityUpdateTransition} stateTransition
+   * @return {Promise<ValidationResult>}
+   */
+  // eslint-disable-next-line no-unused-vars
+  async function validateIdentityUpdateTransitionState(stateTransition) {
+    const result = new ValidationResult();
+
+    const identityId = stateTransition.getIdentityId();
+    const identity = await stateRepository.fetchIdentity(identityId);
+
```

Comment:
I would suggest to clone Identity since you changing it (`new Identity(identity.toObject())`)